src/PostgREST/App.hs

```diff
@@ -102,6 +102,19 @@ app conf reqBody req =
             (cs $ userPass u) (cs $ userRole u)
           return $ responseLBS status201
             [ jsonH
+            , (hLocation,"/postgrest/users?id=eq." <> cs (userId u))
+            ] ""
+
+    (["postgrest", "users"], "PUT") -> do
+      let user = decode reqBody :: Maybe AuthUser
+
+      case user of
+        Nothing -> return $ responseLBS status400 [jsonH] $
+          encode . object $ [("message", String "Failed to parse user.")]
+        Just u -> do
+           _ <- updateUser (cs $ userId u) (cs $ userPass u) (cs $ userRole u)
+           return $ responseLBS status200
+            [ jsonH
```

Comment:
Because the PUT will now apply to the target resource, we won't need to return a Location header.